x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/alerts_context.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  createContext,
+  memo,
+  useContext,
+  useRef,
+  type MutableRefObject,
+  type PropsWithChildren,
+} from 'react';
+import { noop } from 'lodash';
+
+type ToggleColumn = (columnId: string) => void;
+
+/**
+ * Temporary context to share any APIs between the alerts table and other higher level components
+ * such as the alerts details flyout
+ *
+ * TODO remove once the alerts table columns are controllable from the outside
+ */
+export const AlertsContext = createContext<{
+  toggleColumnRef?: MutableRefObject<ToggleColumn | null>;
+}>({
+  toggleColumnRef: undefined,
```

Comment:
`nit` : I think this why Eslint is complaining because value is not pure Ref. We might try to remove this.

Reply:
I removed the effect by directly sharing the table ref (see https://github.com/elastic/kibana/pull/208052#issuecomment-2615246820)